eclass/dotnet-utils.eclass

```diff
@@ -0,0 +1,216 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: dotnet-utils.eclass
+# @MAINTAINER:
+# anna@navirc.com
+# @AUTHOR:
+# Anna Figueiredo Gomes <anna@navirc.com>
+# @SUPPORTED_EAPIS: 7 8
+# @BLURB: common functions and variables for dotnet builds
+
+case ${EAPI} in
+	7|8) ;;
+	*) die "${ECLASS}: EAPI ${EAPI} unsupported." ;;
+esac
+
+if [[ ! ${_DOTNET_UTILS_ECLASS} ]]; then
+_DOTNET_UTILS_ECLASS=1
+
+inherit edo multiprocessing
+
+# @ECLASS_VARIABLE: DOTNET_SLOT
+# @REQUIRED
+# @PRE_INHERIT
+# @DESCRIPTION:
+# Allows to choose a slot for dotnet
+
+if [[ ! ${DOTNET_SLOT} ]]; then
+	die "${ECLASS}: DOTNET_SLOT not set"
+fi
+
+# Temporary, use the virtual once you can have multiple virtuals installed at once
+BDEPEND+=" || ( dev-dotnet/dotnet-sdk-bin:${DOTNET_SLOT} dev-dotnet/dotnet-sdk:${DOTNET_SLOT} )"
+
+IUSE+=" debug"
+
+# @ECLASS_VARIABLE: DOTNET_EXECUTABLE
+# @INTERNAL
+# @DESCRIPTION:
+# Sets to the valid path of a dotnet executable
+
+# @ECLASS_VARIABLE: DOTNET_OUTPUT
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Path of the output directory, if unset, is set during src_compile to:
+# "${WORKDIR}"/${P}_net${DOTNET_SLOT}_${DOTNET_CONFIGURATION}/${PN}
+
+# @ECLASS_VARIABLE: DOTNET_CONFIGURATION
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Configuration value passed to dotnet in the compile phase.
+# Defaults to Debug or Release, depending on the debug useflag.
+
+# @ECLASS_VARIABLE: DOTNET_CLI_TELEMETRY_OPTOUT
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Disables telemetry on dotnet.
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+
+# @ECLASS_VARIABLE: MSBUILDDISABLENODEREUSE
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Stops the dotnet node after the build is done.
+export MSBUILDDISABLENODEREUSE=1
+
+# @ECLASS_VARIABLE: DOTNET_NOLOGO
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Disables the header logo when running dotnet commands.
+export DOTNET_NOLOGO=1
+
+# Needed otherwise the binaries may break
+RESTRICT+=" strip"
```

Comment:
also QA_..., see csharp-language-server ebuild.

Reply:
It's probably right to not strip them vs QA_* warn about prestripping.

Reply:
It's `QA_PREBUILT=".*"`, not "prestripped".

Reply:
Added the ´QA_PREBUILD=".*"´